src/dune_engine/rules.ml

```diff
@@ -47,9 +46,12 @@ module Dir_rules = struct
         | Alias { name; spec } -> Right (name, spec))
     in
     let aliases =
-      Alias.Name.Map.of_list_multi aliases
-      |> Alias.Name.Map.map ~f:(fun specs ->
-        List.fold_left specs ~init:Alias_spec.empty ~f:Alias_spec.union)
+      let add_item what = function
+        | None -> Some what
+        | Some base -> Some (Alias_spec.union what base)
+      in
+      List.fold_left aliases ~init:Alias.Name.Map.empty ~f:(fun acc (name, item) ->
+        Alias.Name.Map.update acc name ~f:(add_item item))
```

Comment:
This change seems unnecessary too?

Reply:
It's a performance improvement. `Alias.Name.Map.of_list_multi` does work to maintain the order (`List.rev` at the end) and this code avoids it.

Reply:
I see. I'm just concerned that by combining multiple improvements, we are going to misunderstand the performance characteristics of the new appendable list implementation.

Reply:
It's possible to isolate this change, but then it's going to add a bunch of spurious changes to the tests. If you think it's worth it, I can do it.

Reply:
Wait, are you saying this change is undoing the list reversal due to `to_list_rev`?

I didn't realise that. At minimum, this deserves a comment, similar to the one you added in `src/dune_engine/load_rules.ml`.

Reply:
(And then maybe we could keep it indeed.)

Reply:
> Wait, are you saying this change is undoing the list reversal due to to_list_rev?

Yes. This is why there are no spurious changes in the tests now.

> I didn't realise that. At minimum, this deserves a comment, similar to the one you added in src/dune_engine/load_rules.ml.

Added a comment.

Reply:
Thanks!